apps/passport-client/src/dispatch.ts

```diff
@@ -323,8 +332,17 @@ async function finishLogin(user: User, state: AppState, update: ZuUpdate) {
 // Runs periodically, whenever we poll new participant info.
 async function setSelf(self: User, state: AppState, update: ZuUpdate) {
   let userMismatched = false;
+  let hasChangedPassword = false;
 
-  if (BigInt(self.commitment) !== state.identity.commitment) {
+  if (state.self && self.salt !== state.self.salt) {
```

Comment:
cc @artwyman

Reply:
I don't quite understand how this is able to be robust to you changing your password in a different tab

Reply:
When we change the password, we update the "self" in localstorage, which contains the salt. On refresh, we have a `loadInitialState()` function that loads the redux state from localStorage. When other tabs reload their pages after one tab changes password, they load the updated "self" from localstorage, which means they don't end up in this `anotherDeviceChangedPassword` state. Also confirmed this experimentally.